reana_server/scheduler.py

```diff
@@ -12,17 +12,70 @@
 import logging
 
 from bravado.exception import HTTPBadGateway, HTTPNotFound
+from kubernetes.client.rest import ApiException
+from sqlalchemy.exc import SQLAlchemyError
+
+from reana_commons.config import REANA_MAX_CONCURRENT_BATCH_WORKFLOWS
 from reana_commons.consumer import BaseConsumer
-from reana_commons.tasks import reana_ready
-from reana_db.database import Session
-from reana_db.models import Workflow
+from reana_commons.k8s.api_client import current_k8s_corev1_api_client
+from reana_db.models import Workflow, WorkflowStatus
 
 from reana_server.api_client import (
     current_rwc_api_client,
     current_workflow_submission_publisher,
 )
 
 
+def check_predefined_conditions():
+    """Check Kubernetes predefined conditions for the nodes."""
+    try:
+        node_info = json.loads(
+            current_k8s_corev1_api_client.list_node(
+                _preload_content=False
+            ).data.decode()
+        )
+        for node in node_info["items"]:
+            # check based on the predefined conditions about the
+            # node status: MemoryPressure, OutOfDisk, KubeletReady
+            #              DiskPressure, PIDPressure,
+            for condition in node.get("status", {}).get("conditions", {}):
+                if not condition.get("status"):
+                    return False
```

Comment:
Thanks for the observation. If things were working as they seem the solution is almost correct, however, this is actually broken because what we get back from the response is not a boolean, it is a string, yielding always `True` when calling `condition.get("status")`, see:
```
$ kubectl exec -ti reana-server-765b6b6577-qk5sr bash
> ipython
In [1]: from reana_commons.k8s.api_client import current_k8s_corev1_api_client                                                                                                                                                        
                                                                                                                                                                                                                                      
In [2]: import json                                                                                                                                                                                                                   
                                                                                                                   
In [3]: node_info = json.loads(                                                                                                                                                                                                       
   ...:             current_k8s_corev1_api_client.list_node(                                                       
   ...:                 _preload_content=False                                                                     
   ...:             ).data.decode()                                                                                
   ...:         )                 
...
In [8]: node_info["items"][0]["status"]["conditions"][0]                                                           
Out[8]:                                                                                                            
{'type': 'MemoryPressure',                                                                                         
 'status': 'False',                                                                                                
 'lastHeartbeatTime': '2021-03-09T15:51:01Z',                                                                      
 'lastTransitionTime': '2021-03-09T13:13:54Z',                                                                     
 'reason': 'KubeletHasSufficientMemory',                                                                           
 'message': 'kubelet has sufficient memory available'}       
```
The condition check is more complex than it seems since the [conditions](https://kubernetes.io/docs/concepts/architecture/nodes/#condition) can be of positive nature (e.g. `Ready`, were we would expect `.get("status"` to be `True`) or negative nature (e.g. `DiskPressure`, where we would expect `.get("status"` to be `True`).

I think we should tackle this in a different issue since the PR at hand tackles https://github.com/reanahub/reana-commons/issues/250.